frontend/src/pages/InsideStudyMode.js

```diff
@@ -1,29 +1,59 @@
-import React, { useState } from "react";
+import React from "react";
 import Header from "../main-components/Header";
 import Sidebar from "../main-components/Sidebar";
 import StudyModeContent from "../main-components/StudyModeContent";
+import { startQuiz } from "../main-components/StudyModeGameHandler";
 import css from "./template.css";
+import queryString from "query-string";
 
-function InsideStudyMode(props) {
-  // Handles mobile menu button and updates sidebar view
-  const [sideSetting, setSideSetting] = useState("f");
-  const handleClick = (e) => {
+class InsideStudyMode extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDataFetched: false,
+      rounds: 0,
+      sideSetting: "f",
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  // Initializes a new game
+  async componentDidMount() {
+    const parameters = queryString.parse(props.location.search);
+    const rounds = await startQuiz(parameters.folderKey);
+    this.setState({ rounds, isDataFetched: true });
+  }
+
+  handleClick(event) {
     if (sideSetting === "f") {
-      setSideSetting("t");
+      this.setState({ sideSetting: "t" });
     } else {
-      setSideSetting("f");
+      this.setState({ sideSetting: "f" });
     }
   };
 
-  return (
-    <div className='App'>
-      <Header id='head' handleClick={handleClick}></Header>
-      <div id='main'>
-        <Sidebar bool={sideSetting}></Sidebar>
-        <StudyModeContent></StudyModeContent>
+  render() {
+    // Handles mobile menu button and updates sidebar view
+    if (!this.state.isDataFetched) {
+      return (
+        <div className='App'>
+          <Header id='head' handleClick={this.handleClick}></Header>
+          <div id='main'>
+            <Sidebar bool={this.state.sideSetting}></Sidebar>
+          </div>
+        </div>
+      );
+    }
+    return (
+      <div className='App'>
+        <Header id='head' handleClick={this.handleClick}></Header>
+        <div id='main'>
+          <Sidebar bool={this.state.sideSetting}></Sidebar>
```

Comment:
bool is not a good name. Maybe `visible`?

Reply:
done.